### PR TITLE
Added new intro mission Scientist Rescue

### PR DESCRIPTION
**Pull request by OcelotWalrus**

### CODE REVIEW

**Review by JovialFeline — COMMENTED** [review]
Neat-o. I ran it a few times, had a read, and left some notes.

I am curious about your method for completing this, since mine felt cheesy. My pods had a rough time, so I force-attacked the ground with Drone Ships to get vision, then re-targeted Bunkers with those drones until the pods could move beyond the beach.

I also inferred from our discussion on the discord that the evacuation ship would reject orders, but I don't see that rule change here. It's small potatoes either way, since the scientist's "removal" triggers victory. Cheers.

**Comment by OcelotWalrus** [comment]
> I am curious about your method for completing this, since mine felt cheesy. My pods had a rough time, so I force-attacked the ground with Drone Ships to get vision, then re-targeted Bunkers with those drones until the pods could move beyond the beach.

Yeah that's what you're supposed to do, before disembarking.

> I also inferred from our discussion on the discord that the evacuation ship would reject orders, but I don't see that rule change here. It's small potatoes either way, since the scientist's "removal" triggers victory. Cheers.

Yeah that rules that did that were causing trouble, and I didn't want to annoy myself with that so I just removed it. `It's small potatoes either way, since the scientist's "removal" triggers victory.` indeed.

**Comment by OcelotWalrus** [comment]
I'll walk into your suggestions later today.

File: mods/hv/maps/scientist-rescue/rules.yaml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+		Intensity: 0.45
+	CloudSpawner:
+		Image: cloud
+		Sequences: cloud1, cloud1f, cloud2, cloud2f, cloud3, cloud3f
+		Palette: cloud
+		Speed: 20, 30
+	LuaScript:
+		Scripts: common|scripts/utils.lua, hv|scripts/campaign.lua, scientist-rescue.lua
+	MissionData:
+		Briefing: A Yuruki scientist working a new technology of sub-spatial televator transportation was kidnapped by the military forces of Synapol. The Yuruki information services were able to retrace the scientist, and you and your crews are now heading to the Synapol research basement located in the surroundings of New Boston's shores. Break into the basement, find the scientist and protect him until the rescue dropship comes to get you.
```
[the remark this answers is not in the document]
Comment (OcelotWalrus) [reply mods/hv/maps/scientist-rescue/rules.yaml:23]: Indeed!

Comment (OcelotWalrus) [inline mods/hv/maps/scientist-rescue/rules.yaml:23]:
```suggestion
		Briefing: A Yuruki scientist working a new technology of sub-spatial televator transportation was kidnapped by the military forces of Synapol. The Yuruki information services were able to retrace the scientist, and you and your crews are now heading to the Synapol research center located in the surroundings of New Boston's shores. Break into the basement, find the scientist and protect him until the rescue dropship comes to get you.
```

File: mods/hv/maps/scientist-rescue/scientist-rescue.lua [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+		Human.MarkCompletedObjective(RescueScientistObjective)
+	end)
+end
+
+SetUnitStances = function()
+	Utils.Do(Map.NamedActors, function(a)
+		if a.Owner == Human then
+			a.Stance = "Defend"
+		end
+	end)
```
[the remark this answers is not in the document]
Comment (OcelotWalrus) [reply mods/hv/maps/scientist-rescue/scientist-rescue.lua:82]: There was originally a Missile Launcher Submarine, that by default has the ReturnFire stance, but I eventually removed it.

File: mods/hv/maps/scientist-rescue/scientist-rescue.lua [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+InsertionPath = { EntryPoint.Location, ShorePoint.Location }
+PassengerReinforcements = { "mgpod", "mgpod", "electricpod", "electricpod", "mortarpod", "mortarpod", "flamepod" }
+OpeningAttack = { Ship1, Ship2, Ship3, Ship4 }
+
+
+Warning = UserInterface.Translate("warning")
+
+SendInsertionNavalTransport = function()
+	local passengers = Reinforcements.ReinforceWithTransport(Human, "navaltransport", PassengerReinforcements, InsertionPath)[2]
+	Trigger.OnAllKilled(passengers, RescueFailed)
```
Comment (JovialFeline) [inline mods/hv/maps/scientist-rescue/scientist-rescue.lua:21]:
```suggestion
	Trigger.OnAllKilled(passengers, function()
		if not Human.IsObjectiveCompleted(KillGuardsObjective) then
			RescueFailed()
		end
	end)
```
If the ship and scientist have spawned, I wouldn't mind decoupling their death and mission failure. I will defer to your intent.

File: mods/hv/maps/scientist-rescue/rules.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+		Intensity: 0.45
+	CloudSpawner:
+		Image: cloud
+		Sequences: cloud1, cloud1f, cloud2, cloud2f, cloud3, cloud3f
+		Palette: cloud
+		Speed: 20, 30
+	LuaScript:
+		Scripts: common|scripts/utils.lua, hv|scripts/campaign.lua, scientist-rescue.lua
+	MissionData:
+		Briefing: A Yuruki scientist working a new technology of sub-spatial televator transportation was kidnapped by the military forces of Synapol. The Yuruki information services were able to retrace the scientist, and you and your crews are now heading to the Synapol research center located in the surroundings of New Boston's shores. Break into the basement, find the scientist and protect him until the rescue dropship comes to get you.
```
[the remark this answers is not in the document]
Comment (OcelotWalrus) [reply mods/hv/maps/scientist-rescue/rules.yaml:23]: I don't why but I must have forgotten replacing that. I though remember doing it.

**Review by Mailaender — APPROVED** [review]

**Comment by Dzierzan** [comment]
This tower (the right one) is not "captured" when my units are next to it.
image

**Comment by Dzierzan** [comment]
> Neat-o. I ran it a few times, had a read, and left some notes.
> 
> I am curious about your method for completing this, since mine felt cheesy. My pods had a rough time, so I force-attacked the ground with Drone Ships to get vision, then re-targeted Bunkers with those drones until the pods could move beyond the beach.
> 
> I also inferred from our discussion on the discord that the evacuation ship would reject orders, but I don't see that rule change here. It's small potatoes either way, since the scientist's "removal" triggers victory. Cheers.

I did exactly the same with drone ships to get some vision.


Anyway, if `Yuruki Industries` faction is chosen a playable one for this map, why do I have mortar or flame pods? There's no consistency here.

Replace them with generic pods and electric ones please.

Moreover, I'd also add some enemy patroling machine gun pods, up to 3 to give that mission a bit more life than again not doing anything AI.

Now... I would add a technician as a medic here. However to spice thing up a bit, you could make it depending on the chosen difficulty. Like Easy - with techncian, Medium and Hard - without one. But that's up to debate.

**Comment by Dzierzan** [comment]
Ships have `Attack Anything Stance` by default, it shouldn't be like that.

**Comment by OcelotWalrus** [comment]
> This tower (the right one) is not "captured" when my units are next to it. image

I have no idea why.

**Comment by OcelotWalrus** [comment]
> Anyway, if Yuruki Industries faction is chosen a playable one for this map, why do I have mortar or flame pods? There's no consistency here.

Yeah I thought mortar and flame were Yuruki's.

**Comment by OcelotWalrus** [comment]
> Moreover, I'd also add some enemy patroling machine gun pods, up to 3 to give that mission a bit more life than again not doing anything AI.

That'd be nice indeed

**Comment by OcelotWalrus** [comment]
> Now... I would add a technician as a medic here. However to spice thing up a bit, you could make it depending on the chosen difficulty. Like Easy - with techncian, Medium and Hard - without one. But that's up to debate.

Personally I wouldn't add a technician. It'd make the mission drastically easier.

**Comment by OcelotWalrus** [comment]
The new commit removes the "AttackAnything" stance to the player's units, makes Yuruki's units Yuruki's units, adds Synapol troops patrolling along the beach, and finally adds a technician to the Yuruki's troops. (I added the technician because of the patrolling troops making it almost impossible to pass through the beach)

**Comment by Dzierzan** [comment]
It's better now!

Move `Naval Transport` a bit up, it's right now on the edge of the map, it doesn't look so great.

Remove land critter, they are responsible for "capturing" watch towers.

Remove ability to capture stuff for `Technician`.

Wrong `Generator`, it's YI's, not SC's.
image

**Comment by Dzierzan** [comment]
Also, make some use of `Facing` in map editor. You always place actors facing the default direction.

image

**Comment by OcelotWalrus** [comment]
Updated with your new suggestions @Dzierzan.

**Comment by Dzierzan** [comment]
Alright, considering the critter issue will be resolved in another PR, I think this mission is good enough.

**Comment by OcelotWalrus** [comment]
I still removed the critters anyway.

**Pull request merged**